jerry-core/parser/js/js-parser-internal.h

```diff
@@ -296,6 +296,59 @@ typedef struct
 
 #endif /* JERRY_DEBUGGER */
 
+#ifndef CONFIG_DISABLE_ES2015_MODULE_SYSTEM
+/**
+ * String struct for storing the module name.
+ */
+typedef struct parser_module_utf8_string
+{
+  uint8_t *value_p;     /**< string value stored as uint8_t */
+  prop_length_t length; /**< length of the string */
+} parser_module_utf8_string_t;
+
+/**
+ * Imported or exported names, for example:
+ * import { v as x } from "module";
+ *  exported name: v
+ *  imported name: x
+ * export x as v;
+ *  exported variable: x
+ *  exported as: v
+ * Reference: https://www.ecma-international.org/ecma-262/6.0/#table-41
+ */
+typedef struct parser_module_names
+{
+  parser_module_utf8_string_t import_name; /**< local name of the import - export item */
+  parser_module_utf8_string_t local_name;  /**< import name of the import - export item */
```

Comment:
Is this a local->import map? If yes, what about using objects for this (for faster property search).

Reply:
Thanks for your review, I’ve updated the patch.

`import { a as b } from 'module.js';`

If a specific module (`module.js`) has been run, its lexical environment is being checked. If that lexical environment contains a property with the given name (`a`), then a new property is created in the parent context with the imported name (`b`) and assigned with the value of `a`.

If objects are being used for the local name -> import name mapping, the key should be the module path. This object should contain as many key-value pairs as many import items were in the import statement. 
![module_map](https://user-images.githubusercontent.com/16573361/53647705-63b5ab80-3c3e-11e9-8a44-d534bf28d01c.png)

I think in this scenario objects won’t worth it, because we now have 1 property search per import item (_in the lexical environment_), then we should have min. 2 (_in the object too_).